R/lazyframe__lazy.R

```diff
@@ -1952,3 +1952,60 @@ LazyFrame_group_by_dynamic = function(
   ) |>
     unwrap("in $group_by_dynamic():")
 }
+
+#' Plot the query plan
+#'
+#' Note that this requires to have the package `DiagrammeR` installed. Use
+#' `raw_output = TRUE` to obtain the raw GraphViz code without plotting it.
+#'
+#' @param ... Additional arguments passed to `DiagrammeR::grViz()`.
+#' @param optimized Optimize the query plan.
+#' @param raw_output Return the raw GraphViz code only, does not print the plot.
+#' @inheritParams LazyFrame_set_optimization_toggle
+#'
+#' @return If `raw_output = TRUE`, a character vector. Otherwise, an object of
+#' class `"grViz"`.
+#'
+#' @examplesIf requireNamespace("DiagrammeR", quietly = TRUE)
+#' query = pl$LazyFrame(mtcars)$
+#'   filter(pl$col("drat") > 3)$
+#'   with_columns(foo = pl$col("mpg") + pl$col("cyl"), bar = pl$mean("mpg"))
+#'
+#' query$show_graph(raw_output = TRUE)
+#'
+#' query$show_graph()
+LazyFrame_show_graph = function(
+    ...,
+    optimized = TRUE,
+    raw_output = FALSE,
+    type_coercion = TRUE,
+    predicate_pushdown = TRUE,
+    projection_pushdown = TRUE,
+    simplify_expression = TRUE,
+    slice_pushdown = TRUE,
+    comm_subplan_elim = TRUE,
+    comm_subexpr_elim = TRUE,
+    streaming = FALSE) {
+  lf = self$set_optimization_toggle(
+    type_coercion,
+    predicate_pushdown,
+    projection_pushdown,
+    simplify_expression,
+    slice_pushdown,
+    comm_subplan_elim,
+    comm_subexpr_elim,
+    streaming
+  ) |> unwrap("in $show_graph():")
+
+  graph = .pr$LazyFrame$to_dot(self, optimized) |>
+    unwrap("in $show_graph():")
+
+  if (isTRUE(raw_output)) {
+    return(graph)
+  }
+
+  if (!requireNamespace("DiagrammeR", quietly = TRUE)) {
+    stop('The package "DiagrammeR" is required.')
+  }
+  DiagrammeR::grViz(graph, ...)
```

Comment:
I am not sure if it is a good idea to pull this package into a dependency here.
I think the situation is different in R and Python in that the user simply needs to run `<LazyFrame>$show_graph() |> DiagrammeR::grViz()`, and there are several other packages with similar functionality.

Reply:
It would be great if there was a way to use `ggplot2` to get a static output (and not an `htmlwidget`), especially since we already suggests `ggplot2`

Reply:
`Rgraphviz` and `DiagrammeR` are listed here.
<https://graphviz.org/resources/>

Since it is easy to chain functions by pipe operators in R, unlike Python, I feel it would be better to simply implement `$to_dot()` and let the user create the plot in a function of their choice.

Reply:
Yes, agreed